File: java-main/java-main/DSA/Array/p32.java
class Demo{
        public static void main(String[]args){
                int N=4;
                int i=0;
                int j=0;
                int arr[][]=new int[][]{{1,2,3,4},{5,6,7,8},{9,10,11,12},{13,14,15,16}};
                while(N>1){
			for(int x=0;x<N-1;x++){
                        	System.out.println(arr[i][j]);
                        	j++;
                	}
                	for(int x=0;x<N-1;x++){
                        	System.out.println(arr[i][j]);
                        	i++;
                	}
                	for(int x=0;x<N-1;x++){
                        	System.out.println(arr[i][j]);
                       		j--;
                	}
                	for(int x=0;x<N-1;x++){
                        	System.out.println(arr[i][j]);
                        	i--;
			}
			N=N-2;
			i++;
			j++;
		}
		if(N==1)
			System.out.println(arr[i][j]);
	}
}

File: bootcamp_c_cpp_dsa-main/bootcamp_c_cpp_dsa-main/CPP/codes/p32.cpp
#include<iostream>
class Demo{
	int x=10;
	static int y;

	public:
	void fun(){
		int a=10;
		static int b=20;
		std::cout<< x <<std::endl;
		std::cout<< y <<std::endl;
		std::cout<< a <<std::endl;
		std::cout<< b <<std::endl;
	}
	static void gun(){
	int a=10;
	static int b=20;
//	std::cout<< x <<std::endl;
	std::cout<< y <<std::endl;
	std::cout<< a <<std::endl;
	std::cout<< b <<std::endl;
	}
};
int main(){
	Demo obj1;
	int Demo::y=20;
	Demo *obj2=new Demo();
	obj1.fun();
	obj1.gun();
	obj2->fun();
	obj2->gun();
	return 0;
}

